src/ConfigWidgets/SymbolLocationsDialog.cpp

```diff
@@ -258,38 +275,64 @@ void SymbolLocationsDialog::OnAddFileButtonClicked() {
 
 ErrorMessageOr<void> SymbolLocationsDialog::TryAddSymbolFile(
     const std::filesystem::path& file_path) {
-  // If the dialog was opened without a module, every valid symbols file with build id can be added
-  if (!module_.has_value()) {
-    OUTCOME_TRY(CheckValidSymbolsFileWithBuildId(file_path));
-    return TryAddSymbolPath(file_path);
-  }
+  {  // Additional scope for add_file_metric timing
+    ScopedMetric add_file_metric{metrics_uploader_, OrbitLogEvent::ORBIT_SYMBOL_LOCATIONS_ADD_FILE};
+    // If the dialog was opened without a module, every valid symbols file with build id can be
+    // added
+    if (!module_.has_value()) {
+      ErrorMessageOr<void> check_result = CheckValidSymbolsFileWithBuildId(file_path);
+      if (check_result.has_error()) {
+        add_file_metric.SetStatusCode(OrbitLogEvent::INTERNAL_ERROR);
+        return check_result;
+      }
+      return TryAddSymbolPath(file_path, std::move(add_file_metric));
```

Comment:
Same here, instead of passing metric to `TryAddSymbolPath`, set the metric status in `TryAddSymbolFile` according to the result return from `TryAddSymbolPath`.

Reply:
done